lib/BaseController.js

```diff
@@ -19,11 +19,25 @@ const clone = require('clone');
 const merge = require('deepmerge');
 const fs = require('fs-extra');
 const hash = require('object-hash');
-
+const BOOL_EXPR = /^(true|false)$/;
 
 module.exports = class BaseController {
   constructor(params) {
 
+    let reconcileByDefault = 'true';
+
+    if (objectPath.has(params, 'reconcileByDefault')) {
+      const valid = BOOL_EXPR.test(params.reconcileByDefault);
+      if (!valid) {
+        throw new TypeError(
+          `BaseController.reconcileByDefault must be a boolean value. Got: ${params.reconcileByDefault}`
+        );
+      }
+
+      reconcileByDefault = String(params.reconcileByDefault);
```

Comment:
interested in your thoughts.. do you think it beneficial to move this `reconcileByDefault` into some kind of `options` object (ie. `params.options.reconcileByDefault`)? My thinking being this leaves open the posibility to add more environment variable configuration in the future if necessary without shoving them all in the params layer, since the params layer has to be updated inside EventHandler. Just want to avoid having to modify 3 files in the future if possible.

Reply:
@alewitt2 updated. Is this more of what you were thinking?

Reply:
yes! hopefully will make our lives easier in the future. thanks